plugin/db/mysql/dump.go

```diff
@@ -226,26 +234,57 @@ func dumpTxn(ctx context.Context, txn *sql.Tx, database string, out io.Writer, s
 			}
 		}
 
+		// Disable foreign key check.
+		// mysqldump uses the same mechanism. When there is any schema or data dependency, we have to disable
+		// the unique and foreign key check so that the restoring will not fail.
+		if _, err := io.WriteString(out, disableUniqueAndForeignKeyCheckStmt); err != nil {
+			return err
+		}
+
 		// Table and view statement.
+		// We have to dump the table before views because of the structure dependency.
 		tables, err := getTablesTx(txn, dbName)
 		if err != nil {
 			return errors.Wrapf(err, "failed to get tables of database %q", dbName)
 		}
+		// Construct temporal views.
```

Comment:
Needs to better explain what issue does temporal view address. By looking at the code this generated CREATE TABLE statement seems invalid as it misses the FROM tbl statement.

Reply:
Done. Referenced mysqldump.cc as well.

Reply:
Construct temporary views.

Reply:
I feel the original mysqldump comment is incorrect.

"Create a table" -> "Create a temporal view"

"The table will be removed" -> "The temporal view"